### PR TITLE
Added random single-qubit CUE and Clifford gates

### PR DESCRIPTION
**Pull request by gauravgyawali**
Working from Seneca's last PR that provided `class RandomizedMeasurements`, I
1. added an additional argument to the `__call__` function to include `unitary_ensemble: str = "pauli"` which can be set to either `"pauli"`, `"clifford"` or `"cue"`
2. added functions for generating random single-qubit Clifford and cue gates. 
3. replaced `unitaries_to_moment` function with `random_single_qubit_unitary_moment` to incorporate all three unitary ensembles instead of just Pauli

### CODE REVIEW

File: cirq-core/cirq/transformers/randomized_measurements.py
Comment (NoureldinYosri) [inline cirq-core/cirq/transformers/randomized_measurements.py:92]: nit: resolve `rng` here so instead of within the other functions

File: cirq-core/cirq/transformers/randomized_measurements.py
Comment (NoureldinYosri) [inline cirq-core/cirq/transformers/randomized_measurements.py:99]: what does `cue` stand for?

**Review by NoureldinYosri — COMMENTED** [review]
just an initial review

File: cirq-core/cirq/transformers/randomized_measurements.py
Comment (gauravgyawali) [reply cirq-core/cirq/transformers/randomized_measurements.py:103]: circular unitary ensemble. added the full form in documentation

File: cirq-core/cirq/transformers/randomized_measurements.py
Comment (gauravgyawali) [reply cirq-core/cirq/transformers/randomized_measurements.py:96]: done.

File: cirq-core/cirq/transformers/randomized_measurements.py
Comment (senecameeks) [inline cirq-core/cirq/transformers/randomized_measurements.py:65]: nit: python style guide recommends [avoiding abbreviations](go/pystyle#naming)

s/subsys_qubits/subsystem_qubits

File: cirq-core/cirq/transformers/randomized_measurements.py
Comment (senecameeks) [inline cirq-core/cirq/transformers/randomized_measurements.py:62]: nit: consider adding `unitary_ensemble = lower(unitary_ensemble)` to add support for users passing "Pauli" etc...

**Review by senecameeks — COMMENTED** [review]
LGTM % a few nits
I'm OOO today so don't wait on my approval :) 

Thanks Gaurav for improving this transformer!

File: cirq-core/cirq/transformers/randomized_measurements.py
Comment (gauravgyawali) [reply cirq-core/cirq/transformers/randomized_measurements.py:63]: done

File: cirq-core/cirq/transformers/randomized_measurements.py
Comment (gauravgyawali) [reply cirq-core/cirq/transformers/randomized_measurements.py]: done

**Review by senecameeks — APPROVED** [review]
LGTM % passing coverage CI (just add one more test), and signing google contributor agreement.

Thanks for working on this!

**Pull request merged**
